src/RegExLib.Core/Entities/Expression.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using RegExLib.Core.Events;
+using RegExLib.Core.Interfaces;
+using RegExLib.SharedKernel;
+
+namespace RegExLib.Core.Entities
+{
+    public class Expression : BaseEntity
+    {
+        public int AuthorId { get; private set; }
+        public string Title { get; }
+        public string Pattern { get; }
+        public string Description { get; }
+        public Author Author { get; private set; }
+
+        public Expression(string title, string pattern, string description)
+        {
+            Title = title;
+            Pattern = pattern;
+            Description = description;
+        }
+
+        public void SetAuthor(Author author)
+        {
+            Author = author;
+            AuthorId = author.Id;
+        }
+
+        public override string ToString()
+        {
+            return string.IsNullOrEmpty(Title) ?string.Empty: Title;
```

Comment:
Same feedback here regarding NRTs.

Also, probably we want some tests. I don't go for 100% test coverage but I do try to test everywhere there is logic, meaning usually conditional statements. Looking at the code so far, you can test:

- ToString when value is null vs. not
- Null checks in various methods
- @ replacement works as expected both when username includes @ and when it does not

Reply:
@ardalis What's the purpose of the `ToString` methods? I don't think I've ever overridden it. Is this best practice for Entities?

I have an [open question regarding the @ replacement](https://github.com/ardalis/RegExLib.com/pull/14#discussion_r447090278): `string` vs `Username` VO.

Reply:
By default ToString will just give the type. If it would be more useful to see something else (usually it is), you can override it. If we're following YAGNI it's probably premature to have it since we don't yet have anything calling it that I know of, but aside from that it's probably fine.

Reply:
There's no issue with using a value object with EF. You can use Owned Entities to configure them. I'm not sure the username really needs one, but it wouldn't be awful.

I tend to try and use PDD - Pain Driven Development - when deciding whether or not to use patterns, including VO. In this case I don't see what pain it's alleviating. (PDD and YAGNI are close cousins)